traffic_ops/app/lib/API/Deliveryservice.pm

```diff
@@ -964,17 +964,24 @@ sub get_deliveryservices_by_userId {
 		#no access to resource tenant
 		return $self->forbidden("Forbidden. User tenant is not available to the working user.");
 	}
-	my $user_ds_ids = $self->db->resultset('DeliveryserviceTmuser')->search( { tm_user_id => $user_id } );
 
+	my %criteria;
+	if ( !$tenant_utils->use_tenancy() ) {
+		my $user_ds_ids = $self->db->resultset('DeliveryserviceTmuser')->search( { tm_user_id => $user_id } );
+		$criteria{'me.id'} = { -in => $user_ds_ids->get_column('deliveryservice')->as_query };
+	}
 	my $deliveryservices = $self->db->resultset('Deliveryservice')
-		->search( { 'me.id' => { -in => $user_ds_ids->get_column('deliveryservice')->as_query } }, { prefetch => [ 'cdn', 'profile', 'type', 'tenant' ] } );
+		->search( \%criteria, { prefetch => [ 'cdn', 'profile', 'type', 'tenant' ] } );
 
 	my @data;
 	if ( defined($deliveryservices) ) {
 		while ( my $row = $deliveryservices->next ) {
 			if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $row->tenant_id)) {
```

Comment:
would it make sense if is_ds_resource_accessible takes in 3 arguments instead of 2? like this? so you could do this?

is_ds_resource_accessible($tenants_data, $row->tenant_id, $user->tenant_id)

or 

is_ds_resource_accessible($tenants_data, $row->tenant_id, $current_user()->tenant_id)

basically you could pass  in ANY tenant id that you want to filter by...just a thought

Reply:
I prefer have separate functions for readability